### PR TITLE
Reaction observation

### PR DESCRIPTION
**Pull request by Einh06**
**What's in this PR**
In this PR was implemented observation for reactions on message.
There is a subtlety in that every `MessageObserverToken` has to observe every Reaction, because we want to notify when list of user changes for a given reaction.

### CODE REVIEW

File: Source/Model/Message/ConversationMessage+Reaction.swift [hunk 1]
```diff
@@ -32,6 +32,14 @@ extension ZMMessage {
         message.addReaction(unicodeValue, forUser: ZMUser.selfUserInContext(context))
     }
     
+    public static func removeReaction(onMessage message:ZMConversationMessage)
```
Comment (marcoconti83) [inline Source/Model/Message/ConversationMessage+Reaction.swift]: Should we check if the reaction was not here in the first place, before removing it?

File: Source/Notifications/ObjectObserverTokens/MessageObserverToken.swift [hunk 2]
```diff
@@ -138,9 +153,17 @@ public final class MessageObserverToken: ObjectObserverTokenContainer, ZMUserObs
         
         changeHandler = {
             [weak self] (_, changeInfo) in
+            if changeInfo.reactionsChanged {
+                if let strongSelf = self {
+                    strongSelf.createTokensForReactions(Array<Reaction>(strongSelf.observedMessage.reactions))
+                }
+                // add a new the new reaction observer
```
Comment (marcoconti83) [inline Source/Notifications/ObjectObserverTokens/MessageObserverToken.swift]: is this a TODO?

File: Source/Model/Message/ConversationMessage+Reaction.swift (hunk 1 above)
Comment (Einh06) [inline Source/Model/Message/ConversationMessage+Reaction.swift]: If the user is not there, nothing would happen. We would still iterate through all reaction to find it, but i think that's fine.

File: Source/Notifications/ObjectObserverTokens/MessageObserverToken.swift (hunk 2 above)
Comment (Einh06) [inline Source/Notifications/ObjectObserverTokens/MessageObserverToken.swift]: Yes, this is out of date, removing.

**Pull request merged**